Enhancements in clarity to access-application-cluster/connecting-frontend-backend.md

murillodigital opened the pull request:
fixes #25818

murillodigital commented:
@Aut0R3V @sftim I've made changes to the overall document towards more accurate description and better clarity, including renaming the manifests and config files to better match their place inside the tutorial. I do have one question, the nginx config in the docs is already baked inside an already published dockerimage `gcr.io/google-samples/hello-frontend:1.0` the change to service name and related change to match in the nginx config would effectively be a breaking change for anybody applying the new manifests using the existing dockerimage. How do we go about publishing a new version to match the documentation?

tengqm commented on content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md:
```diff
@@ -87,50 +89,59 @@ Events:
 
-At this point, you have a backend Deployment running, and you have a
-Service that can route traffic to it.
+At this point, you have a `backend` Deployment running seven replicas of your `hello`
```
Sounds to me that 7 replicas is an overkill. Can/shall we reduce it to 3?

murillodigital replied:
Agreed, reduced to 3 (here as well as deployment manifest and sample apply output)

tengqm commented on content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md:
```diff
@@ -87,50 +89,59 @@ Events:
-The frontend connects to the backend worker Pods by using the DNS name
-given to the backend Service. The DNS name is "hello", which is the value
-of the `name` field in the preceding Service configuration file.
+Now that you have your backend running, you can create a frontend that is available 
```
"available" may be a little ambiguous?

```suggestion
Now that you have your backend running, you can create a frontend that is accessible from 
```

tengqm commented on content/en/docs/tasks/access-application-cluster/connecting-frontend-backend.md:
```diff
@@ -87,50 +89,59 @@ Events:
+Similar to the backend, the frontend has a Deployment and a Service. An important
+difference to notice between the backend and frontend services, is that the
+configuration for the frontend Service has `type: LoadBalancer`, which means that
+the Service uses the default load balancer of your cloud provider and will be
```
```suggestion
the Service uses a load balancer of your cloud provider and will be
```

tengqm commented:
> the change to service name and related change to match in the nginx config would effectively be a breaking change for anybody applying the new manifests using the existing dockerimage.

Will the new "nginx.conf" override the defaults in the docker image? If not, this problem may become a blocker.

murillodigital commented:
> 
> 
> > the change to service name and related change to match in the nginx config would effectively be a breaking change for anybody applying the new manifests using the existing dockerimage.
> 
> Will the new "nginx.conf" override the defaults in the docker image? If not, this problem may become a blocker.

The Dockerfile does indeed replace the default conf file with our frontend-nginx.conf:

```
RUN rm /etc/nginx/conf.d/default.conf
COPY frontend-nginx.conf /etc/nginx/conf.d
```

 The problem is, the current image will attempt to proxy traffic to a hostname named `hello`, which with the introduced changes no longer exists (the new backend service name is `backend`)

tengqm commented:
> The problem is, the current image will attempt to proxy traffic to a hostname named hello, which with the introduced changes no longer exists (the new backend service name is backend)

In that case, we may need to change the backend service name back to `"hello"`?

murillodigital commented:
> 
> 
> > The problem is, the current image will attempt to proxy traffic to a hostname named hello, which with the introduced changes no longer exists (the new backend service name is backend)
> 
> In that case, we may need to change the backend service name back to `"hello"`?

Either that or replace the current image or update one with a newer version, which is what I would prefer.

I don't think the name `hello` is ideal to properly communicate what the hostname is pointing at, nor consistent with the rest of the nomenclature (`backend` vs `frontend` vs `app=hello`).

My preferred approach is build/push a new image `gcr.io/google-samples/hello-frontend:1.1`  to match the changes in this PR. and update the deployment manifest accordingly - for that, I would need somebody with push access to the google-samples gcr repository.

tengqm commented:
> My preferred approach is build/push a new image gcr.io/google-samples/hello-frontend:1.1 to match the changes in this PR. 

Doable, but we need to find the person for the image upgrade and we will need to hold this PR till the new image is published.

murillodigital commented:
> 
> 
> > My preferred approach is build/push a new image gcr.io/google-samples/hello-frontend:1.1 to match the changes in this PR.
> 
> Doable, but we need to find the person for the image upgrade and we will need to hold this PR till the new image is published.

Patience is a virtue 🙂. How would I escalate/get a hold of somebody with access and cycles to support? Is there something other than delay that I may be overlooking?

tengqm commented:
Just tried to google the source for this image but failed. From gcr, I can see the image was created in 2016 and never updated since its initial version. Anyway, I'm okay with upgrading the image too, so long the whole example works.

tengqm commented:
Is this the source code? https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/tree/master/hello-app

murillodigital commented:
> 
> 
> Is this the source code? https://github.com/GoogleCloudPlatform/kubernetes-engine-samples/tree/master/hello-app

You don't need to change the actual hello-app, only the nginx frontend. The Dockerfile to build it is in the docs repo itself, it's nothing but an nginx image with a swapped config file.

The Dockerfile is at `/content/en/examples/service/access` - you'd just have to build it from that path with a tag `gcr.io/google-samples/hello-frontend:1.1` and push it.

tengqm commented:
@murillodigital I see. Thanks.

murillodigital commented:
> 
> 
> @murillodigital I see. Thanks.

Hi there @tengqm ! Just doing a quick drive by to see if you had been able to make some progress on the image building? Let me know if there's anything I can do to help out. Hope you have a phenomenal weekend!

tengqm commented:
@murillodigital Unfortunately, I don't have access to the image repo.

murillodigital commented:
> 
> 
> @murillodigital Unfortunately, I don't have access to the image repo.

Not a problem, do you think it's reasonably possible to find somebody with access, or would you advise that, although less than ideal from my perspective, we simply revert the service name (all other changes would still apply)?

tengqm commented:
@murillodigital I don't know who has this access. Maybe keep the service name as `hello` is a simpler alternative.

murillodigital commented:
> 
> 
> @murillodigital I don't know who has this access. Maybe keep the service name as `hello` is a simpler alternative.

Cool, just renamed the backend service back to `hello` and updated references for consistency, fixed some wording around the term looking to keep clarity - there are no blockers to merging anymore.

murillodigital commented:
@tengqm  - quick ping on this, this should be good to go with my last change.

tengqm commented:
/label tide/merge-method-squash

tengqm commented:
/lgtm
/approve
